packages/sitecore-jss/src/personalize/layout-personalizer.ts

```diff
@@ -13,7 +20,11 @@ export function personalizeLayout(layout: LayoutServiceData, segment: string): v
   });
 }
 
-function personalizePlaceholder(
+/**
+ * @param {Array} components components within placeholder
+ * @param {string} segment segmentId
+ */
+export function personalizePlaceholder(
   components: Array<ComponentRendering | HtmlElementRendering>,
   segment: string
 ): Array<ComponentRendering | HtmlElementRendering> {
```

Comment:
Is it possible to rewrite `for` loop to use `newComponents = components.map`? It would look better